FungibleToken.test.ts

```diff
@@ -538,41 +583,50 @@ describe("token integration", () => {
       await tx.send()
 
       equal(
-        (await vanillaContract.getBalanceOf(sender)).toBigInt(),
+        (await tokenBContract.getBalanceOf(sender)).toBigInt(),
         initialBalanceSender - sendAmount.toBigInt(),
       )
       equal(
-        (await vanillaContract.getBalanceOf(receiver)).toBigInt(),
+        (await tokenBContract.getBalanceOf(receiver)).toBigInt(),
         initialBalanceReceiver + sendAmount.toBigInt(),
       )
+      equal(
+        (await tokenBContract.getCirculating()).toBigInt(),
+        initialCirculating,
+      )
     })
 
-    it("should not allow changing the total supply for token B", async () => {
+    it("should not mint too many B tokens", async () => {
+      FungibleToken.adminContract = CustomTokenAdmin
       await rejects(async () =>
         await Mina.transaction({
           sender: sender,
           fee: 1e8,
         }, async () => {
-          await tokenBContract.setSupply(mintAmount)
+          await tokenBContract.mint(sender, illegalMintAmount)
         })
       )
+      FungibleToken.adminContract = FungibleTokenAdmin
     })
-    it("should not allow changing supply using the vanilla admin contract", async () => {
-      const vanillaContract = new FungibleToken(tokenB)
+    it("should not mint too many B tokens using the vanilla admin contract", {
+      skip: !proofsEnabled,
+    }, async () => {
```

Comment:
why this is skipped when proofs are disabled?

Reply:
Good question! I observed that this test fails when proofs are disabled.

My suspicion is that when proofs are disabled, the verification key of the admin contract is never checked, so you can get away with switching the admin contract. And this test tests that that is prevented (which only works when the proofs are actually constructed and checked).

Reply:
Right, that makes sense